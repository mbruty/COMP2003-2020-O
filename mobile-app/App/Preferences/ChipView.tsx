import React, { useState } from "react";
import { Text, View, StyleSheet } from "react-native";
import { Chip } from "./ChipView/Chip";
interface Props {
  chipNameList: Array<string>;
}
export const ChipView: React.FC<Props> = (props) => {
  const [enabledArray, setEnabledArray] = useState<Array<boolean>>(
    new Array<boolean>(props.chipNameList.length).fill(true)
  );
  const setTouched = (touched: boolean, index: number) => {
    let enabled = [...enabledArray];
    enabled[index] = touched;
    setEnabledArray(enabled);
  };
  return (
    <View style={styles.container}>
      {props.chipNameList.map((chip, index) => (
        <Chip
          title={chip}
          enabled={enabledArray[index]}
          index={index}
          setTouched={setTouched}
        />
      ))}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "flex-start",
    marginLeft: 20,
    flexWrap: "wrap"
  }
})
