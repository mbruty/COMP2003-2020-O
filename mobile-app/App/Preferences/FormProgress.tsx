import React from "react";
import {
  Dimensions,
  ImageSourcePropType,
  StyleSheet,
  Text,
  TouchableOpacity,
  TouchableOpacityComponent,
  View,
} from "react-native";
import { CONSTANT_STYLES } from "../constants";
import { Icon } from "./FormProgress/Icon";

// ToDo: If this is ever needed any where else, maybe add this to the props?
const items: Array<{
  text: string;
  img: { active: ImageSourcePropType; inactive: ImageSourcePropType };
}> = [
  {
    text: "Create Your \nAccount",
    img: {
      active: require("./FormProgress/account_circle_active.png"),
      inactive: require("./FormProgress/account_circle_inactive.png"),
    },
  },
  {
    text: "Confirm Your \nEmail",
    img: {
      active: require("./FormProgress/email_active.png"),
      inactive: require("./FormProgress/email_inactive.png"),
    },
  },
  {
    text: "Select Your \nPreferences",
    img: {
      active: require("./FormProgress/food_active.png"),
      inactive: require("./FormProgress/food_inactive.png"),
    },
  },
];

interface Props {
  selectedIdx: number;
}
export const FormProgress: React.FC<Props> = (props) => {
  const dimensions = Dimensions.get("window");
  const imageWidth = dimensions.width;
  return (
    <>
      <View style={[styles.container, { width: imageWidth - 25 }]}>
        {items.map((x, index) => (
          <>
            <Icon
              text={x.text}
              image={
                index === props.selectedIdx ? x.img.active : x.img.inactive
              }
            />

            {index < items.length - 1 && <View style={styles.line} />}
          </>
        ))}
      </View>
      <View style={[styles.container, { marginTop: -15 }]}>
        <TouchableOpacity
          style={[
            props.selectedIdx === 0
              ? CONSTANT_STYLES.BG_LIGHT_GREY
              : CONSTANT_STYLES.BG_DARK_GREY,
            styles.btn,
            { marginRight: 150 },
          ]}
        >
          <Text style={[CONSTANT_STYLES.TXT_BASE, styles.txt]}>BACK</Text>
        </TouchableOpacity>
        <TouchableOpacity style={[CONSTANT_STYLES.BG_RED, styles.btn]}>
        <Text style={[CONSTANT_STYLES.TXT_BASE, styles.txt]}>{props.selectedIdx === 2 ? "SUBMIT" : "NEXT"}</Text>
        </TouchableOpacity>
      </View>
    </>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "space-between",
    alignSelf: "center",
    padding: 15,
    paddingTop: 25,
  },
  line: {
    height: 2,
    backgroundColor: "#707070",
    width: 50,
    alignSelf: "center",
    marginTop: -35,
    flexShrink: 1,
  },
  btn: {
    paddingLeft: 25,
    paddingRight: 25,
    paddingTop: 10,
    paddingBottom: 10,
    borderRadius: 10,
  },
  txt: {
    fontWeight: "bold",
  },
});
