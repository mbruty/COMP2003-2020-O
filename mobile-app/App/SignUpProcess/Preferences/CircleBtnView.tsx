import React from "react";
import { StyleSheet, View } from "react-native";
import { CircleBtn } from "./CircleBtnView/CircleBtn";

interface Props {
  options: Array<string>;
  touchedArray: Array<boolean>;
  setTouched: (value: boolean, index: number) => void;
}

export const CircleBtnView: React.FC<Props> = (props) => {
  return (
    <View style={styles.container}>
      {props.options.map((option, index) => (
        <CircleBtn
          setTouched={props.setTouched}
          name={option}
          selected={props.touchedArray[index]}
          index={index}
          key={`circle-btn-${index}`}
        />
      ))}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "row",
    flexWrap: "wrap",
    justifyContent: "flex-start",
    paddingLeft: 20,
    paddingTop: 5,
    paddingBottom: 0
  },
});
