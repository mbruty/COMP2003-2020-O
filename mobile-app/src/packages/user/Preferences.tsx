import { AntDesign, Ionicons } from "@expo/vector-icons";
import React, { useState } from "react";
import {
  Modal,
  ScrollView,
  StyleSheet,
  Text,
  TouchableOpacity,
  TouchableWithoutFeedback,
  View,
} from "react-native";
import { CONSTANT_COLOURS, CONSTANT_STYLES } from "../../constants";
import useToggle from "../../hooks/useToggle";
import AwesomeAutoCompleteInput from "../controls/AwesomeAutoCompleteInput";

interface Props {}

const values = ["Burger", "Beans", "Big Booty Bitches"];
let timeout;

const Preferences: React.FC<Props> = (props) => {
  const [value, setValue] = useState<string>("");

  const [showModal, toggleModal] = useToggle(false);

  const debounceInput = (text: string, delay: number = 500) => {
    clearTimeout(timeout);
    timeout = setTimeout(
      () =>
        // Make API Call here
        console.log(text),
      delay
    );
    setValue(text);
  };

  return (
    <ScrollView contentContainerStyle={styles.container}>
      <Modal animationType="fade" transparent={true} visible={showModal}>
        <TouchableWithoutFeedback onPress={toggleModal}>
          <View style={styles.centeredView}>
            <TouchableWithoutFeedback onPress={() => {}}>
              <View style={styles.modalView}>
                <TouchableOpacity
                  onPress={toggleModal}
                  style={{
                    alignSelf: "flex-end",
                    marginTop: -15,
                  }}
                >
                  <AntDesign name="closecircleo" size={24} color="#AAA" />
                </TouchableOpacity>
                <Text style={{marginTop: -15}}>
                  We won't show you any reccomendations that contains these tags
                </Text>
              </View>
            </TouchableWithoutFeedback>
          </View>
        </TouchableWithoutFeedback>
      </Modal>
      <AwesomeAutoCompleteInput
        autoComplete={values}
        value={value}
        setValue={debounceInput}
      />
      <Text style={[styles.txt, CONSTANT_STYLES.TXT_DEFAULT]}>
        I like to eat
      </Text>
      <View style={styles.seperator} />
      <View style={{ alignSelf: "flex-start" }}>
        <TouchableOpacity
          style={styles.txtContainer}
          onPress={() => toggleModal()}
        >
          <Text style={[styles.txt, CONSTANT_STYLES.TXT_DEFAULT]}>
            I don't like to eat
          </Text>
          <Ionicons
            name="ios-information-circle-outline"
            size={24}
            color={CONSTANT_COLOURS.DARK_GREY}
            style={{ paddingTop: 5, paddingLeft: 5 }}
          />
        </TouchableOpacity>
      </View>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
  },
  centeredView: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    marginTop: 22,
    backgroundColor: "rgba(0, 0, 0, 0.4)",
  },
  modalView: {
    margin: 20,
    backgroundColor: "white",
    borderRadius: 20,
    padding: 35,
    alignItems: "center",
    shadowColor: "#000",
    width: "80%",
    shadowOffset: {
      width: 0,
      height: 2,
    },
  },
  txtContainer: {
    flexDirection: "row",
    alignSelf: "flex-start",
    alignItems: "center",
  },
  txt: {
    paddingLeft: 25,
    paddingVertical: 15,
    fontSize: 18,
    alignSelf: "flex-start",
  },
  seperator: {
    width: "90%",
    borderColor: "#AAAAAA",
    borderWidth: 1,
    borderRadius: 5,
  },
});
export default Preferences;
