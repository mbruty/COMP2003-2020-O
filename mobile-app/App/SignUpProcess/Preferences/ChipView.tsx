import React, { useState } from "react";
import { Text, View, StyleSheet } from "react-native";
import { Chip } from "./ChipView/Chip";
interface Props {
  chipNameList: Array<string>;
  enabledArray: Array<boolean>;
  setTouched: (touched: boolean, index: number) => void;
}
export const ChipView: React.FC<Props> = (props) => {
  

  return (
    <View style={styles.container}>
      {props.chipNameList.map((chip, index) => (
        <Chip
          key={`chip-${index}`}
          title={chip}
          enabled={props.enabledArray[index]}
          index={index}
          setTouched={props.setTouched}
        />
      ))}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "flex-start",
    marginLeft: 20,
    flexWrap: "wrap",
  },
});
