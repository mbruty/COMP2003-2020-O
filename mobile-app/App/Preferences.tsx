import React, { useState } from "react";
import {
  Dimensions,
  Image,
  ScrollView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from "react-native";
import { CircleBtnView } from "./Preferences/CircleBtnView";
import { ChipView } from "./Preferences/ChipView";
import { CONSTANT_STYLES } from "./constants";
import { FormProgress } from "./Preferences/FormProgress";

// ToDo: Fetch this list from the API?

const allergies = ["Lactose", "Nuts", "Gluten", "Soy", "Shellfish"];
const typesOfFood = ["Fish", "Vegan", "Vegetarian", "Meat", "Halal", "Kosher"];

interface Props {
  fName: string;
  setPage?: React.Dispatch<React.SetStateAction<string>>;
  onSubmit: () => void;
}

export const Preferences: React.FC<Props> = (props) => {
  const [allergiesBoolArr, setAllergiesBoolArr] = useState<Array<boolean>>(
    new Array<boolean>(allergies.length).fill(true)
  );

  const [foodBoolArr, setFoodBoolArr] = useState<Array<boolean>>(
    new Array<boolean>(allergies.length).fill(true)
  );

  /* Set the allergies array to the value at the index
    This will be used by the chips to update the state of the form on
    a touch event from a chip 
  */
  const setAllergies = (touched: boolean, index: number) => {
    let enabled = [...allergiesBoolArr];
    enabled[index] = touched;
    setAllergiesBoolArr(enabled);
  };

  // Same as above, but for food types
  const setFoodTypes = (touched: boolean, index: number) => {
    let enabled = [...foodBoolArr];
    enabled[index] = touched;
    setFoodBoolArr(enabled);
  };

  // Called on submit button click
  const submit = () => {
    // ToDo:
    // Send preferences to the server...
    props.setPage("main");
  };

  // The react-native way of doing width: 100vw; height: 100vh;
  const dimensions = Dimensions.get("window");
  const imageHeight = Math.round((dimensions.width * 9) / 16);
  const imageWidth = dimensions.width;

  const textStyle = [styles.text, CONSTANT_STYLES.TXT_DEFAULT];
  return (
    <ScrollView>
      <View style={{ elevation: 25 }}>
        <Image
          style={{
            height: imageHeight,
            width: imageWidth,
            marginTop: -15,
          }}
          source={require("./Preferences/preferences_banner.png")}
        />
      </View>
      <Text
        allowFontScaling={false}
        style={[styles.bannerText, CONSTANT_STYLES.TXT_BASE]}
      >
        Hi {props.fName}!{"\n"}
        What types of food can you{"\n"}
        eat?
      </Text>
      <Text
        allowFontScaling={false}
        style={[textStyle, { marginTop: 0, marginBottom: 0 }]}
      >
        Types of food
      </Text>
      <CircleBtnView
        options={typesOfFood}
        touchedArray={foodBoolArr}
        setTouched={setFoodTypes}
      />
      <Text allowFontScaling={false} style={[textStyle, { marginTop: 15 }]}>
        Allergies / intolerance
      </Text>
      <ChipView
        enabledArray={allergiesBoolArr}
        setTouched={setAllergies}
        chipNameList={allergies}
      />
      <FormProgress onSubmit={props.onSubmit} allowBack={false} selectedIdx={2} />
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  text: {
    margin: 15,
    marginTop: 0,
    fontSize: 25,
    fontWeight: "bold",
  },
  bannerText: {
    position: "absolute",
    top: 35,
    left: 15,
    fontSize: 30,
    fontWeight: "bold",
    elevation: 26,
  },
  submit: {
    alignSelf: "flex-end",
    marginRight: 40,
    marginTop: 40,
  },
  submitText: {
    fontSize: 15,
    fontWeight: "bold",
  },
});
