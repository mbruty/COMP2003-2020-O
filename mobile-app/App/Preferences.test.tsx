import React, { useState } from "react";
import renderer from "react-test-renderer";
import { Preferences } from "./Preferences";

test("Renders Circle Button List", () => {
  const preferecnesPage = renderer
    .create(<Preferences fName="Test" />)
    .toJSON();
  expect(preferecnesPage).toMatchSnapshot();
});
