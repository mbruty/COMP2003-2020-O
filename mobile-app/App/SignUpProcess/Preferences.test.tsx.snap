// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Circle Button List 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "elevation": 25,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 422,
            "marginTop": -15,
            "width": 750,
          }
        }
      />
    </View>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "elevation": 26,
            "fontSize": 30,
            "fontWeight": "bold",
            "left": 15,
            "position": "absolute",
            "top": 35,
          },
          Object {
            "color": "#F1F1F1",
          },
        ]
      }
    >
      Hi 
      Test
      ,
      

      what types of food can you
      

      eat?
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Array [
            Object {
              "fontSize": 25,
              "fontWeight": "bold",
              "margin": 15,
              "marginTop": 0,
            },
            Object {
              "color": "#707070",
            },
          ],
          Object {
            "marginBottom": 0,
            "marginTop": 0,
          },
        ]
      }
    >
      Types of food
    </Text>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "flex-start",
          "paddingBottom": 0,
          "paddingLeft": 20,
          "paddingTop": 5,
        }
      }
    >
      <View
        style={
          Object {
            "marginRight": 10,
            "marginTop": 20,
            "width": 80,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#FD4040",
              "borderRadius": 100,
              "elevation": 10,
              "height": 60,
              "opacity": 1,
              "width": 60,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 32,
                "marginLeft": 14,
                "marginTop": 14,
                "width": 32,
              }
            }
          />
        </View>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "marginTop": 5,
                "textAlign": "center",
              },
              Object {
                "color": "#707070",
              },
            ]
          }
        >
          Fish
        </Text>
      </View>
      <View
        style={
          Object {
            "marginRight": 10,
            "marginTop": 20,
            "width": 80,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#FD4040",
              "borderRadius": 100,
              "elevation": 10,
              "height": 60,
              "opacity": 1,
              "width": 60,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 32,
                "marginLeft": 14,
                "marginTop": 14,
                "width": 32,
              }
            }
          />
        </View>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "marginTop": 5,
                "textAlign": "center",
              },
              Object {
                "color": "#707070",
              },
            ]
          }
        >
          Vegan
        </Text>
      </View>
      <View
        style={
          Object {
            "marginRight": 10,
            "marginTop": 20,
            "width": 80,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#FD4040",
              "borderRadius": 100,
              "elevation": 10,
              "height": 60,
              "opacity": 1,
              "width": 60,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 32,
                "marginLeft": 14,
                "marginTop": 14,
                "width": 32,
              }
            }
          />
        </View>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "marginTop": 5,
                "textAlign": "center",
              },
              Object {
                "color": "#707070",
              },
            ]
          }
        >
          Vegetarian
        </Text>
      </View>
      <View
        style={
          Object {
            "marginRight": 10,
            "marginTop": 20,
            "width": 80,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#FD4040",
              "borderRadius": 100,
              "elevation": 10,
              "height": 60,
              "opacity": 1,
              "width": 60,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 32,
                "marginLeft": 14,
                "marginTop": 14,
                "width": 32,
              }
            }
          />
        </View>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "marginTop": 5,
                "textAlign": "center",
              },
              Object {
                "color": "#707070",
              },
            ]
          }
        >
          Meat
        </Text>
      </View>
      <View
        style={
          Object {
            "marginRight": 10,
            "marginTop": 20,
            "width": 80,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#FD4040",
              "borderRadius": 100,
              "elevation": 10,
              "height": 60,
              "opacity": 1,
              "width": 60,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 32,
                "marginLeft": 14,
                "marginTop": 14,
                "width": 32,
              }
            }
          />
        </View>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "marginTop": 5,
                "textAlign": "center",
              },
              Object {
                "color": "#707070",
              },
            ]
          }
        >
          Halal
        </Text>
      </View>
      <View
        style={
          Object {
            "marginRight": 10,
            "marginTop": 20,
            "width": 80,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#FD4040",
              "borderRadius": 100,
              "elevation": 10,
              "height": 60,
              "opacity": 1,
              "width": 60,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 32,
                "marginLeft": 14,
                "marginTop": 14,
                "width": 32,
              }
            }
          />
        </View>
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "marginTop": 5,
                "textAlign": "center",
              },
              Object {
                "color": "#707070",
              },
            ]
          }
        >
          Kosher
        </Text>
      </View>
    </View>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Array [
            Object {
              "fontSize": 25,
              "fontWeight": "bold",
              "margin": 15,
              "marginTop": 0,
            },
            Object {
              "color": "#707070",
            },
          ],
          Object {
            "marginTop": 15,
          },
        ]
      }
    >
      Allergies / intolerance
    </Text>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "flex-start",
          "marginLeft": 20,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 50,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 20,
                "elevation": 10,
                "flex": 1,
                "flexDirection": "row",
                "margin": 5,
                "padding": 8,
                "paddingRight": 15,
              },
              Object {
                "backgroundColor": "#FD4040",
              },
            ]
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 24,
                "marginLeft": 5,
                "width": 24,
              }
            }
          />
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "marginTop": 2,
                  "paddingLeft": 5,
                },
                Object {
                  "color": "#FFF",
                },
              ]
            }
          >
            Lactose
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 50,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 20,
                "elevation": 10,
                "flex": 1,
                "flexDirection": "row",
                "margin": 5,
                "padding": 8,
                "paddingRight": 15,
              },
              Object {
                "backgroundColor": "#FD4040",
              },
            ]
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 24,
                "marginLeft": 5,
                "width": 24,
              }
            }
          />
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "marginTop": 2,
                  "paddingLeft": 5,
                },
                Object {
                  "color": "#FFF",
                },
              ]
            }
          >
            Nuts
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 50,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 20,
                "elevation": 10,
                "flex": 1,
                "flexDirection": "row",
                "margin": 5,
                "padding": 8,
                "paddingRight": 15,
              },
              Object {
                "backgroundColor": "#FD4040",
              },
            ]
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 24,
                "marginLeft": 5,
                "width": 24,
              }
            }
          />
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "marginTop": 2,
                  "paddingLeft": 5,
                },
                Object {
                  "color": "#FFF",
                },
              ]
            }
          >
            Gluten
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 50,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 20,
                "elevation": 10,
                "flex": 1,
                "flexDirection": "row",
                "margin": 5,
                "padding": 8,
                "paddingRight": 15,
              },
              Object {
                "backgroundColor": "#FD4040",
              },
            ]
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 24,
                "marginLeft": 5,
                "width": 24,
              }
            }
          />
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "marginTop": 2,
                  "paddingLeft": 5,
                },
                Object {
                  "color": "#FFF",
                },
              ]
            }
          >
            Soy
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 50,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 20,
                "elevation": 10,
                "flex": 1,
                "flexDirection": "row",
                "margin": 5,
                "padding": 8,
                "paddingRight": 15,
              },
              Object {
                "backgroundColor": "#FD4040",
              },
            ]
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 24,
                "marginLeft": 5,
                "width": 24,
              }
            }
          />
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "marginTop": 2,
                  "paddingLeft": 5,
                },
                Object {
                  "color": "#FFF",
                },
              ]
            }
          >
            Egg
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "row",
            "padding": 15,
            "paddingTop": 25,
          },
          Object {
            "alignSelf": "center",
            "justifyContent": "space-between",
            "width": 725,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 35,
              "width": 35,
            }
          }
        />
        <Text
          allowFontScaling={false}
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Create Your 
Account
        </Text>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#707070",
            "flexShrink": 1,
            "height": 2,
            "marginTop": -35,
            "width": 50,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 35,
              "width": 35,
            }
          }
        />
        <Text
          allowFontScaling={false}
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Confirm Your 
Email
        </Text>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#707070",
            "flexShrink": 1,
            "height": 2,
            "marginTop": -35,
            "width": 50,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 35,
              "width": 35,
            }
          }
        />
        <Text
          allowFontScaling={false}
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Select Your 
Preferences
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "row",
            "padding": 15,
            "paddingTop": 25,
          },
          Object {
            "paddingBottom": 25,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#FD4040",
            "borderRadius": 10,
            "bottom": 0,
            "opacity": 1,
            "paddingBottom": 8,
            "paddingLeft": 20,
            "paddingRight": 20,
            "paddingTop": 8,
            "position": "absolute",
            "right": 25,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#F1F1F1",
              },
              Object {
                "fontWeight": "bold",
              },
            ]
          }
        >
          SUBMIT
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
